core/deployment/src/main/java/io/quarkus/deployment/test/TestDefiningAnnotationBuildItemProvider.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.deployment.test;
+
+import org.jboss.builder.BuildStep;
+import org.jboss.builder.item.BuildItem;
+import org.jboss.jandex.DotName;
+
+// this is only needed because the test modules do not depend on Arc
+public interface TestDefiningAnnotationBuildItemProvider {
```

Comment:
I don't know if we really need this SPI, as it is not really usable outside this specific case.

We could probably just do:


----
    @BuildStep
    BeanDefiningAnnotationBuildItem testBeans(LaunchModeBuildItem launchModeBuildItem) {
        if(launchModeBuildItem.getLaunchMode() == LaunchMode.TEST) {
            return new BeanDefiningAnnotationBuildItem(DotName.createSimple("io.quarkus.test.junit.QuarkusTest"));
        }
        return null;
    }
----

For Junit 4 you could also declare @RunWith as a bean defining annotation.

In some ways its a bit less 'pure' as you have the test annotation names in the ArC module however it is a lot simpler and should be a lot more obvious what is going on.

Reply:
Sounds good to me! I will go ahead and make the changes you propose.